### ISSUE TITLE
Cannot modify $allowed_children

### ISSUE THREAD

**Issue by JorisDebonnet** [silverstripe/silverstripe-blog#391]
I want to limit the kind of subpages my Blog can have. By default, it can have any Page as subpage, and any BlogPost and its extended classes as items in the Lumberjack gridfield. 

Now, assume I want it to accept only one type of normal page and one type of BlogPost: 

```php
<?php
class Article extends BlogPost {/* custom stuff */}
class Subpage extends Page {}
```
```yaml
Blog:
  allowed_children:
    - Article
    - Subpage
```

Well, that doesn't work, at least not fully: it succesfully restricts the subpages (that will appear in the site tree) to only "Subpage", but although it shows "Article" at the top of the dropdown, it still also shows "BlogPost" (and any additional BlogPost pagetypes) as an option to create new items (in the Lumberjack gridfield). So it seems like the 'original BlogPost' is forced as an allowed child. 

Is there a way around this, or can this be fixed so it can be configured as above? 

I know I can do `class ArticleHolder extends Blog` and succesfully specify a single type of BlogPost-extension as children _there_, but I'm just wondering whether I can achieve that on the "original" Blog class.

**Comment by micmania1** [comment]
I think this is a problem with config merging rather than this module.

You should be able to remove the default `$allowed_children` using `Config::inst()->remove('Blog', 'allowed_children');` and then re-add your allowed children using `Config::inst()->update('Blog', 'allowed_children', ['Article']);`

**Comment by dhensby** [comment]
Yes, to remove items from the config you have to do it in PHP as @micmania1 has said

**Comment by dhensby** [comment]
Alternatively, you could decorate and update canCreate or maybe there's a config option to hide page types from the create screen...

**Comment by JorisDebonnet** [comment]
Mm, well when I try the config method, I still have the same issue: allowing BlogPost as a child automatically also allows all of its extended classes as child. Perhaps it simply can't be avoided. I also looked at canCreate for a short time, but even always returning false from BlogPostExtension::canCreate does not have any effect. 

I decided to give up and stay with my workaround (extending the Blog class).
